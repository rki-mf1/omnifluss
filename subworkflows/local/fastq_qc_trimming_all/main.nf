include { FASTQC                 } from '../../../modules/nf-core/fastqc/main'
include { FASTP                  } from '../../../modules/nf-core/fastp/main'

workflow FASTQ_QC_TRIMMING_ALL {
    take:
    tools     // string
    ch_reads  // channel: [ val(meta), fastq ]

    main:
    ch_trimmed_reads  = Channel.empty()
    ch_multiqc_files  = Channel.empty()
    ch_versions       = Channel.empty()

    // FASTQC
    if (tools.split(',').contains('fastqc')) {
        FASTQC (
            ch_reads
        )
        ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]})
        ch_versions = ch_versions.mix(FASTQC.out.versions.first())
    }

    // FASTP
    if (tools.split(',').contains('fastp')) {
        FASTP(
            ch_reads,
            [],
            false,
            false,
            false
        )
        ch_multiqc_files = ch_multiqc_files.mix(FASTP.out.json.collect{it[1]})
        ch_versions = ch_versions.mix(FASTP.out.versions.first())

        ch_trimmed_reads = FASTP.out.reads
    }

    emit:
    trimmed_reads     = ch_trimmed_reads  // channel: [ val(meta), fastq ]

    multiqc_files     = ch_multiqc_files  // channel: [ multiqc_files ]
    versions          = ch_versions       // channel: [ versions.yml ]

}
