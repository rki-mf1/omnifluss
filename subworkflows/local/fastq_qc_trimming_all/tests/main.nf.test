nextflow_workflow {

    name "Test Subworkflow FASTQ_QC_TRIMMING_ALL"
    script "../main.nf"
    // Incude a nextflow.config for the test if needed
    config "subworkflows/local/fastq_qc_trimming_all/tests/nextflow.config"
    workflow "FASTQ_QC_TRIMMING_ALL"

    //  Add tags for the subworkflow
    tag "subworkflows"
    tag "subworkflows/local"
    tag "subworkflows/fastq_qc_trimming_all"
    tag "fastq_qc_trimming_all"
    // Add tags for all modules used within this subworkflow.
    tag "fastqc"
    tag "fastp"


    // Test name preferably indicating the test-data and file-format used
    test("sarscov2 - single_end - stub") {

        // Activate the stub mode for nextflow
        options '-stub'

        when {
            workflow {
                """
                input[0] = 'fastqc,fastp'
                input[1] = [ [ id:'test', single_end:true ], // meta map
                            [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true) ]
                        ]
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("sarscov2 - paired_end - stub") {

        // Activate the stub mode for nextflow
        options '-stub'

        when {
            workflow {
                """
                input[0] = 'fastqc,fastp'
                input[1] = [ [ id:'test', single_end:false ], // meta map
                            [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true) ]
                        ]
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
