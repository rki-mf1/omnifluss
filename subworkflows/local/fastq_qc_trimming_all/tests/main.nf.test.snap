{
    "sarscov2 - single_end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.fastp.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        "test.fastp.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "test.zip:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,4923a99cb98c366d658eaa38d11e98f0",
                    "versions.yml:md5,988a1086de844c2926d325a53b370b7b"
                ],
                "multiqc_files": [
                    [
                        "test.fastp.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "test.zip:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "trimmed_reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.fastp.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4923a99cb98c366d658eaa38d11e98f0",
                    "versions.yml:md5,988a1086de844c2926d325a53b370b7b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-07-08T16:56:46.290323648"
    },
    "sarscov2 - paired_end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.fastp.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test_2.fastp.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "1": [
                    [
                        "test.fastp.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "test.zip:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,4923a99cb98c366d658eaa38d11e98f0",
                    "versions.yml:md5,988a1086de844c2926d325a53b370b7b"
                ],
                "multiqc_files": [
                    [
                        "test.fastp.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        "test.zip:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "trimmed_reads": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.fastp.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test_2.fastp.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4923a99cb98c366d658eaa38d11e98f0",
                    "versions.yml:md5,988a1086de844c2926d325a53b370b7b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-07-08T16:56:56.511098294"
    }
}